src/daemon/mod.rs

```diff
@@ -141,82 +144,200 @@ pub async fn start_interruptable(opts: CliOpts, config: Config) -> anyhow::Resul
+    db_root_dir: PathBuf,
+    forest_car_db_dir: PathBuf,
+}
+
+fn do_startup_miscs_and_get_timestamp(
+    opts: &CliOpts,
+    config: &Config,
+) -> anyhow::Result<chrono::DateTime<chrono::Utc>> {
+    let start_time = chrono::Utc::now();
+    maybe_increase_fd_limit()?;
+    // Sets proof parameter file download path early, the files will be checked and
+    // downloaded later right after snapshot import step
+    crate::utils::proofs_api::set_proofs_parameter_cache_dir_env(&config.client.data_dir);
+    info!(
+        "Starting Forest daemon, version {}",
+        FOREST_VERSION_STRING.as_str()
+    );
     if opts.detach {
         tracing::warn!("F3 sidecar is disabled in detach mode");
         std::env::set_var("FOREST_F3_SIDECAR_FFI_ENABLED", "0");
     }
+    Ok(start_time)
+}
 
+fn get_chain_config_and_set_network(config: &Config) -> Arc<ChainConfig> {
     let chain_config = Arc::new(ChainConfig::from_chain(config.chain()));
     if chain_config.is_testnet() {
         CurrentNetwork::set_global(Network::Testnet);
     }
+    chain_config
+}
 
-    info!(
-        "Starting Forest daemon, version {}",
-        FOREST_VERSION_STRING.as_str()
-    );
-    maybe_increase_fd_limit()?;
-
-    let start_time = chrono::Utc::now();
-    let path: PathBuf = config.client.data_dir.join("libp2p");
-    let net_keypair = crate::libp2p::keypair::get_or_create_keypair(&path)?;
-    let p2p_peer_id = net_keypair.public().to_peer_id();
-
-    let mut keystore = load_or_create_keystore(&config).await?;
+fn get_or_create_p2p_keypair_and_peer_id(config: &Config) -> anyhow::Result<(Keypair, PeerId)> {
+    let path = config.client.data_dir.join("libp2p");
+    let keypair = crate::libp2p::keypair::get_or_create_keypair(&path)?;
+    let peer_id = keypair.public().to_peer_id();
+    Ok((keypair, peer_id))
+}
 
+async fn load_or_create_keystore_and_configure_jwt(
+    opts: &CliOpts,
+    config: &Config,
+) -> anyhow::Result<(Arc<RwLock<KeyStore>>, String)> {
+    let mut keystore = load_or_create_keystore(config).await?;
     if keystore.get(JWT_IDENTIFIER).is_err() {
         keystore.put(JWT_IDENTIFIER, generate_priv_key())?;
     }
-
-    let admin_jwt = handle_admin_token(&opts, &keystore)?;
-
+    let admin_jwt = handle_admin_token(opts, &keystore)?;
     let keystore = Arc::new(RwLock::new(keystore));
+    Ok((keystore, admin_jwt))
+}
 
-    let chain_data_path = chain_path(&config);
-
+fn migrate_db_if_needed(config: &Config) {
     // Try to migrate the database if needed. In case the migration fails, we fallback to creating a new database
     // to avoid breaking the node.
-    let db_migration = crate::db::migration::DbMigration::new(&config);
+    let db_migration = crate::db::migration::DbMigration::new(config);
     if let Err(e) = db_migration.migrate() {
         warn!("Failed to migrate database: {e}");
     }
+}
 
+async fn prepare_db(opts: &CliOpts, config: &Config) -> anyhow::Result<(Arc<DbType>, DbMetadata)> {
```

Comment:
In general, it might make sense to have uniform naming of the setup methods. I'd see it as something along the lines of `setup_db`, `setup_keystore`, `setup_keypair` with docs on what they do. As noted elsewhere, making the method names overly verbose is not the way to go (especially if you start adding more features to existing ones). I'd make it a bit more high-level, and explain rough details in the function documentation.

Reply:
Fixed.